packages/grid/x-data-grid-premium/src/hooks/features/aggregation/createAggregationLookup.ts

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import {
+  gridColumnLookupSelector,
+  gridFilteredRowsLookupSelector,
+  GridRowId,
+  gridRowIdsSelector,
+  gridRowTreeSelector,
+} from '@mui/x-data-grid-pro';
+import { GridApiPremium } from '../../../models/gridApiPremium';
+import { DataGridPremiumProcessedProps } from '../../../models/dataGridPremiumProps';
+import {
+  GridAggregationFunction,
+  GridAggregationLookup,
+  GridAggregationPosition,
+  GridAggregationRules,
+} from './gridAggregationInterfaces';
+import { getAggregationRules } from './gridAggregationUtils';
+import { gridAggregationModelSelector } from './gridAggregationSelectors';
+
+const getAggregationCellValue = ({
+  apiRef,
+  groupId,
+  field,
+  aggregationFunction,
+  aggregationRowsScope,
+}: {
+  apiRef: React.MutableRefObject<GridApiPremium>;
+  groupId: GridRowId;
+  field: string;
+  aggregationFunction: GridAggregationFunction;
+  aggregationRowsScope: DataGridPremiumProcessedProps['aggregationRowsScope'];
+}) => {
+  const rowTree = gridRowTreeSelector(apiRef);
+  const filteredRowsLookup = gridFilteredRowsLookupSelector(apiRef);
+
+  let rowIds: GridRowId[];
+
+  // TODO: Add custom root id
+  if (groupId === '') {
+    rowIds = gridRowIdsSelector(apiRef).filter((rowId) => !rowTree[rowId].isAutoGenerated);
+  } else {
+    rowIds = apiRef.current.getRowGroupChildren({ groupId });
+  }
+
+  if (aggregationRowsScope === 'filtered') {
+    rowIds = rowIds.filter((rowId) => filteredRowsLookup[rowId] !== false);
+  }
+
+  return aggregationFunction.apply({
+    values: rowIds.map((rowId) => apiRef.current.getCellValue(rowId, field)),
+  });
```

Comment:
When `aggregationRowsScope === 'filtered'` we iterate through `rowIds` twice - first `.filter` and then `.map`.
We can reduce this to one iteration:

```suggestion
  const values: any[] = [];
  rowIds.forEach((rowId) => {
    if (aggregationRowsScope === 'filtered' && filteredRowsLookup[rowId] === false) {
      return;
    }
    values.push(apiRef.current.getCellValue(rowId, field));
  });

  return aggregationFunction.apply({ values });
```

Or using `.reduce`, but for me personally `.forEach` is more clear